marshaller.go

```diff
@@ -279,6 +306,148 @@ func getReflectedValueAndType(v interface{}) (reflect.Value, reflect.Type, error
 	return reflectedValue, reflectedType, nil
 }
 
+// mapSelectColumns maps the selectColumn field name in the soql tag to their
+// corresponding field name in the struct needed by marshalOrderByClause
+func mapSelectColumns(mappings map[string]string, parent string, gusParent string, v interface{}) error {
+	reflectedValue, reflectedType, err := getReflectedValueAndType(v)
+	if err != nil {
+		return ErrInvalidSelectColumnOrderByClause
+	}
+
+	for i := 0; i < reflectedType.NumField(); i++ {
+		field := reflectedType.Field(i)
+		tag := field.Tag.Get(SoqlTag)
+		if tag == "" {
+			continue
+		}
+		// skip all fields that are not tagged as selectColumn
+		if getClauseKey(tag) != SelectColumn {
+			continue
+		}
+
+		fieldName := field.Name
+		gusFieldName := getFieldName(tag, field.Name)
+
+		// inside a nested struct, prepend parent to create full field names
+		if parent != "" {
+			fieldName = parent + period + fieldName
+			gusFieldName = gusParent + period + gusFieldName
+		}
+
+		fieldValue := reflectedValue.Field(i)
+
+		// the mapping for a struct field should be added regardless, to cover
+		// the case of a struct field not being a nested field (e.g. time.Time)
+		mappings[fieldName] = gusFieldName
+		if fieldValue.Kind() == reflect.Struct {
+			err := mapSelectColumns(mappings, fieldName, gusFieldName, fieldValue.Interface())
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+// v is the Order slice for specifying the columns and sort order
+// s is the struct value containing fields with the selectColumn tag
+func marshalOrderByClause(v interface{}, tableName string, s interface{}) (string, error) {
+	reflectedValue, reflectedType, err := getReflectedValueAndType(v)
+	if err != nil {
+		return "", err
+	}
+
+	if reflectedType.Kind() != reflect.Slice {
+		return "", ErrInvalidOrderByClause
+	}
+
+	if reflectedType.Elem() != reflect.TypeOf(Order{}) {
+		return "", ErrInvalidOrderByClause
+	}
+
+	sReflectedValue, sReflectedType, err := getReflectedValueAndType(s)
+	if err != nil {
+		return "", err
+	}
+
+	if sReflectedType.Kind() != reflect.Struct {
+		return "", ErrInvalidSelectColumnOrderByClause
+	}
+
+	columnMappings := make(map[string]string)
+
+	err = mapSelectColumns(columnMappings, "", "", sReflectedValue.Interface())
+	if err != nil {
+		return "", err
+	}
+
+	if len(columnMappings) == 0 {
+		return "", ErrInvalidSelectColumnOrderByClause
+	}
+
+	var buff strings.Builder
+
+	previousConditionExists := false
+	for i := 0; i < reflectedValue.Len(); i++ {
+		order := reflectedValue.Index(i).Interface().(Order)
+		fieldName := order.Field
+		if strings.TrimSpace(fieldName) == "" {
+			return "", ErrInvalidOrderByClause
+		}
+
+		columnName, ok := columnMappings[fieldName]
+		if !ok {
+			return "", ErrInvalidOrderByClause
+		}
+
+		if tableName != "" {
```

Comment:
Wondering when `tableName` will be non empty string? Do we support `orderBy` for nested queries (parent to child relation)?

Reply:
SOQL supports order by clause inside nested query, that is when tableName is non-empty